src/components/root/createModals/CreateEnvironmentModal.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+	Autocomplete,
+	Button,
+	DialogActions,
+	DialogContent,
+	DialogTitle,
+	Divider,
+	FormControl,
+	FormLabel,
+	Input,
+	Modal,
+	ModalDialog,
+} from '@mui/joy';
+import { CreateModalsProps } from './createModalsProps';
+import { Environment, iconFromTabType } from '../../../types/application-data/application-data';
+import { useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { selectEnvironments } from '../../../state/active/selectors';
+import { useAppDispatch } from '../../../state/store';
+import { addNewEnvironment } from '../../../state/active/thunks/environments';
+import { addTabs, setSelectedTab } from '../../../state/tabs/slice';
+
+export function CreateEnvironmentModal({ open, closeFunc }: CreateModalsProps) {
+	const [envName, setEnvName] = useState('');
+	const [cloneFrom, setCloneFrom] = useState<string | null>(null);
+	const allEnvironments = useSelector(selectEnvironments);
```

Comment:
Rather than selecting all environments, there's a selectEnvironmentsById you can use and pass in cloneFrom as the ID.

```
const environmentToCloneFrom = useSelector((state) => selectEnvironmentsById(state, cloneFrom));
```

We can update it to allow null as a value as well and just return null or undefined.

Reply:
but we already need all environments for the clone functionality

Reply:
Yup.

Reply:
I'll resolve this.